hbase-server/src/main/java/org/apache/hadoop/hbase/util/ServerRegionReplicaUtil.java

```diff
@@ -161,25 +162,24 @@ public static void setupRegionReplicaReplication(Configuration conf) throws IOEx
     if (!isRegionReplicaReplicationEnabled(conf)) {
       return;
     }
-    Admin admin = ConnectionFactory.createConnection(conf).getAdmin();
-    ReplicationPeerConfig peerConfig = null;
-    try {
+
+    try (Connection connection = ConnectionFactory.createConnection(conf);
+      Admin admin = connection.getAdmin()) {
+      ReplicationPeerConfig peerConfig = null;
       peerConfig = admin.getReplicationPeerConfig(REGION_REPLICA_REPLICATION_PEER);
-    } catch (ReplicationPeerNotFoundException e) {
-      LOG.warn("Region replica replication peer id=" + REGION_REPLICA_REPLICATION_PEER
-          + " not exist", e);
-    }
-    try {
+
       if (peerConfig == null) {
         LOG.info("Region replica replication peer id=" + REGION_REPLICA_REPLICATION_PEER
-            + " not exist. Creating...");
+          + " not exist. Creating...");
         peerConfig = new ReplicationPeerConfig();
         peerConfig.setClusterKey(ZKConfig.getZooKeeperClusterKey(conf));
         peerConfig.setReplicationEndpointImpl(RegionReplicaReplicationEndpoint.class.getName());
         admin.addReplicationPeer(REGION_REPLICA_REPLICATION_PEER, peerConfig);
       }
-    } finally {
-      admin.close();
+    } catch (ReplicationPeerNotFoundException e) {
+      LOG.warn(
+        "Region replica replication peer id=" + REGION_REPLICA_REPLICATION_PEER + " not exist",
+        e);
```

Comment:
If we catch this Exception, don't we want to create replication peer id?
We can model this with:
```
try(closeables){
  try{
    peerConfig = admin.getReplicationPeerConfig(REGION_REPLICA_REPLICATION_PEER);
  } catch (ReplicationPeerNotFoundException e){
  }
  if (peerConfig == null) {
    .....
  }
}
```

Reply:
yeah, I was about to submit the change for this.